scripts/core/MultiSelectHoc.tsx

```diff
@@ -12,13 +20,24 @@ export interface IMultiSelectOptions {
 
 interface IProps {
     children: (options: IMultiSelectOptions) => JSX.Element;
+
+    /**
+     * When items are updated/deleted, we need to check if they should be unselected
+     * in case they no longer match the query and thus are no longer visible
+     * in the list view.
+     */
+    shouldUnselect(ids: Set<string>): Promise<Set<string>>;
 }
 
-interface Istate {
+interface IState {
     selected: Map<string, IArticle>;
 }
 
-export class MultiSelectHoc extends React.PureComponent<IProps, Istate> {
+export class MultiSelectHoc extends React.PureComponent<IProps, IState> {
+    private removeContentUpdateListener: () => void;
+    private removeResourceDeletedListener: () => void;
+    private maybeUnselectItems: (ids: globalThis.Set<string>) => void;
```

Comment:
doesn't it need the underscore in the name? [this one](https://github.com/superdesk/superdesk-client-core/pull/3722/files?diff=unified&w=1#diff-e7af39c7e776bf103574695df94d8489954718dd1f293fb8eca59173b6770426R47) has it

Reply:
We don't have a convention on this. In a given react component probably 9 of 10 methods and variables would be private so I wouldn't put underscores everywhere, especially when there is a `private` keyword. Regarding `_mounted`, I don't remember why did I add the underscore. Probably because it's a property that doesn't have to do anything with business logic and is just boring and unimportant implementation detail. I would rather drop it there than add underscores to all private members.